docs/source/add_columns.md

```diff
@@ -0,0 +1,111 @@
+# Adding Columns to Metadataset
+
+This document explains how to use the feature to add columns in a Metadataset using the `megatron.energon` module.
+
+Adding columns is analogous to webdataset's add column, i.e., merging two datasets which require the same files within the tar files.
+
+## Minimal Example
+
+Below is a minimal example of a `metadataset_minimal.yaml` file:
+
+```yaml
+__module__: megatron.energon
+__class__: Metadataset
+splits:
+    train:
+        datasets:
+            - weight: 1
+                join:
+                    - path: ds1
+                    - path: ds2
+                join_type:
+                    __module__: megatron.energon
+                    __class__: CaptioningSample
+```
+
+## Extensive Example
+
+Here is a more extensive example of a `metadataset_extended.yaml` file:
+
+```yaml
+__module__: megatron.energon
+__class__: Metadataset
+splits:
+    train:
+        datasets:
+            - weight: 1
+                join:
+                    - path: ds1
+                        dataset_config: dataset.yaml  # If override is needed
+                    - path: ds2
+                        dataset_config: dataset.yaml
+                        subflavor: ds1  # If needed, overrides the 'ds2' subflavor
+                        subflavors: # If needed, will be merged(overriding) with parent subflavor
+                            ds2_extra: 2
+                        split_config: split.yaml  # Sets this for all joined datasets
+                join_type:
+                    __module__: my_module
+                    __class__: JoinedSample # Type should implement from_joined(ds1, ds2)
+                split_config: split.yaml  # Sets this for all joined datasets
+                split_part: train  # Sets this for all joined datasets
+                subflavor: ds1  # Sets this for all joined datasets
+                subflavors:  # Sets this for all joined datasets (it will be merged with their individual subflavors)
+                    source: metadataset.yaml
+```
+
+## Custom Join Type
+
+To define a custom join type, you can create a Python class as shown below in `my_module.py`:
+
+```python
```

Comment:
We need to add more explanation to why the code needs to look like this.

Reply:
Yes, you wanted a basic tutorial :D

Reply:
Yeah, we can keep this task for later.